doc/user.xml

```diff
@@ -0,0 +1,243 @@
+to help you get started quickly in recognising groups with <Package>recog</Package>.
+<P/>
+
+TODO explain sections
+
+<Section Label="background">
+<Heading>Theoretical background</Heading>
+
+At its core, <Package>recog</Package> uses a "divide and conquer" strategy to
+understand the structure of a given group <M>G</M>. This process builds a data
+structure called a <Q>recognition tree</Q> (also referred to as a <Q>composition
+tree</Q>).<P/>
+
+A recognition tree breaks down the group <M>G</M> into smaller, more manageable groups.
+The process typically involves finding a homomorphism <M>\varphi \colon G \to
+H</M>, and then recursively analysing both the kernel
+<M>\ker(\varphi) = N \unlhd G</M> and the image
+im<M>(\varphi) \cong G/N</M>. This continues until the remaining groups
+are (quasi-)simple, which are the fundamental building blocks of all
+groups. The resulting tree represents the entire structure of the original group
+<M>G</M>, where the root is given by <M>G</M> and the leaves are the
+(quasi-)simple groups obtained in the process.
+The leaf nodes are the groups being directly recognised, and the
+result is then assembled back up the tree until the group <M>G</M>
+is recognised.<P/>
+
+In order to work with group elements efficiently and constructively throughout
+this process, 
+<Package>recog</Package> uses <Q>Straight-Line Programs</Q> (or <Q>SLPs</Q>). An <Q>SLP</Q> is a
+highly efficient way to record how a group element <M>g \in G</M> is built from the group's
+generators. Instead of a potentially very long word in terms of the
+generators that evaluates to <M>g</M>,
+an SLP is a short sequence of steps to compute the element <M>g</M>.
+This makes it possible to work with very large groups
+effectively.<P/>
+
+A key problem that <Package>recog</Package> helps to solve is the
+<Q>constructive membership problem</Q>.
+Given a group <M>G = \langle X \rangle</M> by a set of generators <M>X \subseteq
+G</M> and an element <M>g</M>, the problem is twofold. First, is
+<M>g</M> an element of <M>G</M>? Second, if it is, how can <M>g</M> be written as a word in the generators <M>X</M>? The "constructive"
+part of the solution is providing this representation, and <Package>recog</Package> does this by producing an SLP for
+<M>g</M>. This provides a concrete certificate of membership that can be used in
+further computations. Solving this problem efficiently is fundamental to many
+other group algorithms.
+
+</Section>
+
+<Section Label="functions">
+<Heading>Functions</Heading>
+
+The main function to recognise a group is the following.
+
+<#Include Label="RecogniseGroup">
+
+Alternatively, if you know the type of the given group, you can directly call the specialised functions for permutation, matrix, or projective groups.
+
+<#Include Label="RecognisePermGroup">
+<#Include Label="RecogniseMatrixGroup">
+<#Include Label="RecogniseProjectiveGroup">
```

Comment:
Currently the four `Recognize*Group` functions are included in two chapters, GAPDoc prints warnings about that,
I suggest to include `RecognizeGroup` here and to include the other three functions in Chapter "Group recognition".